A few enhancements

pmetras opened the pull request:
This merge requests adds a few enhancements:
- Driver launcher script
- Tablet calibration
- Multi pointer: the mouse on the primary screen and the pencil/stylet on the tablet
- Ubuntu 16.04 LTS example with Huion Kamvas Pro 12" (GT-116) in documentation
- New 5 buttons menu for Gimp and Krita that can be used on Kamvas Pro 12"

pmetras committed:
Driver launcher

pmetras committed:
Add new 5 buttons menu for GT116

pmetras committed:
Add calibrate #49 and multi-pointer

pmetras committed:
Update with GT116 example on Ubuntu

pmetras committed:
Add Ubuntu example into TOC

pmetras committed:
Ubuntu example into TOC take 2

CNB-Kevin commented on huion-tablet-driver.py:
```diff
@@ -259,6 +266,75 @@ def multi_monitor():
+        # Locate output device name
+        xoutput_devices = os.popen("xrandr --listactivemonitors | sort | cut -f 6 -d ' ' ").read()
+        xoutput_devices = list(filter(None, xoutput_devices.split('\n'))) # Assumes the first device is the main monitor and the second is the graphic monitor
+        xoutput_device = xoutput_devices[1]
```
Assuming that there is always 2 monitors is unsafe.
It would be safer to get the last one in the list, but we'd still assume that the last monitor is the graphic monitor.
```suggestion
        xoutput_device = xoutput_devices[-1]
```
The best solution would be to match the monitor with the device, but I'm exactly sure on how to do that...

pmetras replied:
I've integrated a merge request from @shervinsahba that does this, I think.

CNB-Kevin reviewed (commented):
Crash at line 317 of huion-tablet-driver.py when the user only has 1 monitor.

CNB-Kevin commented:
There is an issue where if you turn off the driver (I run it in a terminal and use ctrl+C while I'm testing things) if multi-cursor is enabled (which it is by default). Leftover cursors can stay on screen or appear/disappear.

It could be user error on my part because I turn it off, but it's also caused by the lack of cleanup at the end of the program, which is also a thing on the main branch.

pmetras commented:
Yes, multi-cursor on X is not reliable. There a bit of [cleanup](https://github.com/pmetras/huion-linux-drivers/blob/master/huion-tablet-driver.py#L280) in the code when the driver is started again but it would be better to do it at the end of program, if there were a cleanup procedure before exiting.

pmetras committed:
Use the last display instead of the second

pmetras committed:
Cleanup multi cursor at program exit (NOT TESTED)

Commit:
Create constant XOUTPUT_DEVICE_ARGUMENT to control calibrate_mapping

Introduces a constant so that users can specify their output monitor for calibrate_mapping(). 
This expands on the bug fix mentioned here https://github.com/joseluis/huion-linux-drivers/pull/53#discussion_r362628886

pmetras committed:
Tested multi-pointer cleanup

Commit:
Merge pull request #1 from shervinsahba/master

Create constant XOUTPUT_DEVICE_ARGUMENT to control calibrate_mapping

pmetras commented:
I finally had the opportunity to test the clean up code for multi-cursor (the tablet is 6000 km from my computer!). Leftover cursors when the driver is interrupted shouldn't remain on the screen and X should be more stable now.

slagatoras commented:
hey! first of all thank you for this miracle! everything is ok except i still have the residual pointers in my tablet. also while drawing i get a round semi transparent box around the cursor. and i have two residuals cursors on my tablet after i end the script that keep flickering. aaand while it works there is always one inactive cursor on the side of the tablet. so i have one moving with the mouse.one with the pen and one that just sits there. also even if i set multi cursor to false its still enabled oh sorry 18.04 version!

pmetras commented:
My daughter who owns the tablet told me she has the same behaviour from time to time. She told me that there could be a conflict with Firefox too but I don't have the details. And I can't debug, only when I go and visit my daughter far from home...
The multi-pointer part is really a hack. I hope a temporary one before we have a better driver.

slagatoras commented:
i see well even like that its still great!! but.. sorry really. i upgraded to 20.04 and i have this issue while running the code.would you happen to have any ideas as to why??i'll try to take a look at it tomorow since its past midnight here but i was just wondering!the problem is that the pointer is not directly folowing the pen unless i reach the far right of the screen.
![Screenshot from 2020-04-25 00-31-33](https://user-images.githubusercontent.com/52567506/80258699-81d53b00-868c-11ea-87b2-92694682107f.png)
  edit: it coudl just needed a reboot as now it works flawless . i did change the display settings using LXRandR  but i dont know if its a coincidense! in any case i'll look into it a bit more tomorow

pmetras commented:
Could it be that the calibration of the tablet is not correct?
Or perhaps this pull request https://github.com/joseluis/huion-linux-drivers/pull/62 is correcting your issue?
